pandas/core/groupby/groupby.py

```diff
@@ -489,6 +489,21 @@ def group_selection_context(groupby: "BaseGroupBy") -> Iterator["BaseGroupBy"]:
 class BaseGroupBy(PandasObject, SelectionMixin, Generic[FrameOrSeries]):
     _group_selection: Optional[IndexLabel] = None
     _apply_allowlist: FrozenSet[str] = frozenset()
+    _hidden_attrs = PandasObject._hidden_attrs | {
+        "as_index",
+        "axis",
+        "dropna",
```

Comment:
Do we want these to be hidden? Okay with keeping the same behavior as master in this PR, but maybe an issue should be opened.

Reply:
Good question. From a user perspective, I think it's ok to hide them (helps users focus on groupby methods), but I also think there are situation where we want them to be visible (when developing, advanced use cases).

Maybe store them as not-hidden internal attributes (`_as_index` etc)?